lib/kaffy/resource_callbacks.ex

```diff
@@ -62,6 +62,9 @@ defmodule Kaffy.ResourceCallbacks do
       {:error, :not_found} ->
         Kaffy.Utils.repo().update(changeset)
 
+      {:error, error} ->
+        {:error, error}
```

Comment:
As there's a clause catching "unexpected error" below, the semantic suggests this is an expected error - so maybe just catch `{:error, %Ecto.Changeset{}}` here?

Reply:
it's certainly not expected behavior ;) It's not handled properly in any case:

<img width="806" alt="image" src="https://user-images.githubusercontent.com/6031161/224657606-123fbc10-4fd2-420d-b240-dd064be73098.png">

Reply:
but you made me realise there are more places to handle this (insert and delete).
This for sure could be done better, but I'm not trying to fix the world with this PR, just to make displaying errors work again ;)

Reply:
Yeah, I just wonder if we want to upstream these changes or not. If not then it's probably just fine.

Reply:
so you suggest to just pattern match on changesets? I can do that but the next step is basically always rolling back the transaction so any other error that would be wrapped with double error tuple will fail the same way down the line unless inception errors are handled properly (which is I guess not the best way to handle this :P). The change right now ensures the returned tuple from transaction will not be in the form of `{:error, {:error, ...}}` so even if it's not changeset it has a chance of being handled properly.